web/src/components/MemoEditor.tsx

```diff
@@ -124,17 +125,52 @@ const MemoEditor = () => {
       const contentBeforeCursor = editorRef.current.getContent().slice(0, cursorPosition);
       const rowValue = last(contentBeforeCursor.split("\n"));
       if (rowValue) {
-        if (listItemSymbolList.includes(rowValue)) {
+        if (listItemSymbolList.includes(rowValue) || emptyOlReg.test(rowValue)) {
           event.preventDefault();
           editorRef.current.removeText(cursorPosition - rowValue.length, rowValue.length);
         } else {
+          // unordered list / checked list
+          let matched = false;
           for (const listItemSymbol of listItemSymbolList) {
             if (rowValue.startsWith(listItemSymbol)) {
               event.preventDefault();
               editorRef.current.insertText("", `\n${listItemSymbol}`);
+              matched = true;
               break;
             }
           }
+          if (!matched) {
+            // ordered list
+            const olReg = /^([1-9][0-9]*)\. /;
+            const olRes = olReg.exec(rowValue);
+            if (olRes) {
+              let order = parseInt(olRes[1]) + 1;
+              event.preventDefault();
+              const contentAfterCursor = editorRef.current.getContent().slice(cursorPosition);
+              editorRef.current.insertText("", `\n${order}. `);
+              if (contentAfterCursor) {
+                // correct the order
+                order++;
+                const nextRows = contentAfterCursor.split("\n").slice(1);
+                const rowStart = contentBeforeCursor.split("\n").length + 1;
+                const content = editorRef.current.getContent().split("\n");
+                let updated = false;
+                for (let i = 0; i < nextRows.length; i++) {
+                  const rowRes = olReg.exec(nextRows[i]);
+                  if (rowRes) {
+                    content[rowStart + i] = nextRows[i].replace(rowRes[1], (order + i).toString());
+                    updated = true;
+                  } else {
+                    break;
+                  }
+                }
+                if (updated) {
+                  editorRef.current.setContent(content.join("\n"));
+                  editorRef.current.setCursorPosition(cursorPosition + 4);
+                }
+              }
```

Comment:
I just remove these on my local machine, it still works right. Is there any reason about update cursor position right here?

Reply:
Please see the 15s of the video.

This code is to correct the subsequent number when we enter in the middle of an ordered list.

And setting the cursor position with a hard-coded number is because after I call `setContent` the cursor will be at the end.

Reply:
Maybe the implementation is a bit cumbersome, but I found that there are similar features in github.